dnf5/commands/download/download.hpp

```diff
@@ -36,9 +36,13 @@ class DownloadCommand : public Command {
     explicit DownloadCommand(Command & parent) : Command(parent, "download") {}
     void set_argument_parser() override;
     void configure() override;
+    void goal_resolved() override;
```

Comment:
Do not add this method. It will not be called. Instead, put the code in the `run()` method right after `auto transaction = goal->resolve()`.